quill-jdbc/src/test/scala/io/getquill/JdbcContextSpec.scala

```diff
@@ -0,0 +1,22 @@
+package io.getquill
+
+import java.io.Closeable
+import javax.sql.DataSource
+
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.FlatSpec
+import org.scalatest.MustMatchers
+
+class JdbcContextSpec extends FlatSpec with MustMatchers with MockFactory {
+
+  abstract class DataSourceWithCloseable extends DataSource with Closeable
+
+  "JdbcContextSpec" should "call close when extends Closable" in {
+    val dataSourceMock = mock[DataSourceWithCloseable]
```

Comment:
We avoid mock libraries. Wdyt about making `DataSourceWithCloseable` a concrete class with a counter of `close` calls?